docs/DeveloperGuide.md

```diff
@@ -494,6 +494,52 @@ The below UML diagram represents the overall structure of the predicates for `Bu
 4. Filtering entries by name - that is, providing the `-n` flag to the filter command, will filter all entries whose names
    contain the parameter provided to `-n` as a *substring*.
 
+### Sorting buyers and properties
+
+To sort `Buyers` and `Properties`, the `ObservableArrayList` that stores references to these objects and displays them on the user's screen
+is modified directly to a sorted version. These changes are propagated directly to the `FilteredList`, enabling users to sort
+a previously filtered list. As the `FilteredList` is based on the `ObservableArrayList`, users can also sort the list first,
+then filter it. This results in users being able to build sort and filter functions on top of each other to more powerfully
+manipulate the list based on their needs.
+
+A `Comparator` is used to sort the `ObservableArrayList`. Different comparators with different conditions are used to sort the
+list by different criteria. The following are the `Comparators` used to allow for the corresponding sorting functions:
+`Buyer`: `BuyerComparator`
+1. `NameComparator`
+2. `PriceRange Comparator`
+3. `Priority Comparator`
+`Property`: `PropertyComparator`
+4. `PropertyNameComparator`
+5. `PriceComparator`
+Both:
+6. `TimeComparator`
+
+The UML diagrams below represents the overall structure of the `Comparator`s used.
+
+![BuyerComparatorsClassDiagram](images/SortBuyerComparatorsClassDiagram.png)
+![PropertyComparatorsClassDiagram](images/SortPropComparatorsClassDiagram.png)
+
+#### Design Considerations
+Similar to the `FilteredList` abstraction provided by JavaFX, we considered using a `SortedList` to present the list in a
+sorted version without modifying the underlying data structure `ObservableArrayList`. This is
+to preserve the chronological order in which users enter the entries so that it can still be displayed with the `list` command.
+
+However, this meant that we needed to have both `FilteredList` and `SortedList` stored and vary which one is displayed to users
+on entering a command. As such, the last-shown list changes depending on the last command entered. To keep track of this,
+we used a flag which would be updated everytime a `filter` or `sort` command was used. All `Command`s that referred to an
```

Comment:
Maybe we should explain this as an "overall" thing for both filter and sort?

Reply:
Like after the 2 sections?